Multisig/UI/UI Library/RefreshableScrollView.swift

```diff
@@ -0,0 +1,180 @@
+                }
+            } else {
+                // remove the sapce at the top of the scroll view
+                self.frozen = false
+            }
+
+            // Update last scroll offset
+            self.previousScrollOffset = self.scrollOffset
+        }
+    }
+
+    func symbolRotation(_ scrollOffset: CGFloat) -> Angle {
+        // We will begin rotation, only after we have passed
+        // 60% of the way of reaching the threshold.
+        if scrollOffset < self.threshold * 0.60 {
+            return .degrees(0)
+        } else {
+            // Calculate rotation, based on the amount of scroll offset
+            let h = Double(self.threshold)
+            let d = Double(scrollOffset)
+            let v = max(min(d - (h * 0.6), h * 0.4), 0)
+            return .degrees(180 * v / (h * 0.4))
+        }
+    }
+
+    struct SymbolView: View {
+        let height: CGFloat
+        let loading: Bool
+        let frozen: Bool
+        let rotation: Angle
+
+        var body: some View {
+            Group {
+                if self.loading { // If loading, show the activity control
+                    VStack {
+                        Spacer()
+                        ActivityRep()
+                        Spacer()
+                    }
+                    .frame(height: height).fixedSize()
+                    .offset(y: -height + (self.loading && self.frozen ? height : 0.0))
+                } else {
+                    Image(systemName: "arrow.down") // If not loading, show the arrow
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .frame(width: height * 0.25, height: height * 0.25).fixedSize()
+                        .padding(height * 0.375)
+                        .rotationEffect(rotation)
+                        .offset(y: -height + (loading && frozen ? +height : 0.0))
+                        .foregroundColor(.gray)
+                }
+            }
+        }
+    }
+
+    struct MovingView: View {
+        var body: some View {
+            GeometryReader { proxy in
+                Color.clear.preference(
+                    key: RefreshableKeyTypes.PrefKey.self,
+                    value: [RefreshableKeyTypes.PrefData(vType: .movingView, bounds: proxy.frame(in: .global))])
+            }.frame(height: 0)
+        }
+    }
+
+    struct FixedView: View {
+        var body: some View {
+            GeometryReader { proxy in
+                Color.clear.preference(
+                    key: RefreshableKeyTypes.PrefKey.self,
+                    value: [RefreshableKeyTypes.PrefData(vType: .fixedView, bounds: proxy.frame(in: .global))])
+            }
+        }
+    }
+}
+
+struct RefreshableKeyTypes {
+    enum ViewType: Int {
+        case movingView
+        case fixedView
+    }
+
+    struct PrefData: Equatable {
+        let vType: ViewType
+        let bounds: CGRect
+    }
+
+    struct PrefKey: PreferenceKey {
```

Comment:
same here (unclear name)

Reply:
What would be your proposal on the name here?

Reply:
`Key`?

Reply:
Then `PrefData` would be `Data` which is also not ideal.
I would leave it as it is for now if you don't mind.